cli_tools/common/vdisk/qemu_img.go

```diff
@@ -0,0 +1,107 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vdisk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/files"
+)
+
+// Format is an enum type for representing VM disk encoding formats.
+type Format string
+
+const (
+	// FormatUnknown means that qemu-img could not determine the file's format.
+	FormatUnknown Format = "unknown"
+	// FormatVMDK means that the file uses the vmdk file format.
+	FormatVMDK Format = "vmdk"
+	// FormatVHDX means that the file uses the vhdx file format.
+	FormatVHDX Format = "vhdx"
+	// FormatVPC means that the file uses the vpc file format.
+	FormatVPC Format = "vpc"
+	// FormatVDI means that the file uses the vdi file format.
+	FormatVDI Format = "vdi"
+	// FormatQCOW2 means that the file uses the qcow2 file format.
+	FormatQCOW2 Format = "qcow2"
+	// FormatRAW means that the file uses the raw file format.
+	FormatRAW Format = "raw"
+)
+
+var formats = map[string]Format{
+	"vmdk":  FormatVMDK,
+	"vhdx":  FormatVHDX,
+	"vpc":   FormatVPC,
+	"vdi":   FormatVDI,
+	"qcow2": FormatQCOW2,
+	"raw":   FormatRAW,
+}
```

Comment:
Is this mapping needed? It seems it's 1:1.

Reply:
It provides a whitelist mechanism so that the log can be classified as an enum rather than free-form text.

(It's the only such mechanism that I'm aware of; any alternative ideas?)

Reply:
As mentioned above, maybe just have a set/list of valid values and validate against it (after toLowerCase)?

Reply:
Updated to validate against the list of formats returned by `qemu-img --help`